api/src/main/java/org/apache/flink/agents/api/tool/ToolRegistry.java

```
@@ -0,0 +1,139 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.agents.api.tool;
+
+import org.apache.flink.agents.api.tool.annotation.Tool;
+
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Registry for managing tools available to agents. Works with the merged design where BaseTool
+ * contains ToolMetadata.
+ */
+public class ToolRegistry {
+
+    private final Map<String, BaseTool> tools = new ConcurrentHashMap<>();
+
+    /** Register a tool instance directly. */
+    public void registerTool(BaseTool tool) {
+        String name = tool.getName();
+        if (tools.containsKey(name)) {
+            throw new IllegalArgumentException(
+                    "Tool with name '" + name + "' is already registered");
+        }
+        tools.put(name, tool);
+    }
+
+    /** Register all @Tool annotated methods from an object instance. */
+    public void registerToolsFromObject(Object toolProvider) {
```

Maybe we can only support static method as FunctionTool at first to simplify this pr, and add other support later if necessary.

---

Ok, but Do we want to throw an exception if someone tries to register non-static @Tool methods? or should we only look for @Tool on static methods in the initial implementation?